include/gridtools/common/int_vector.hpp

```diff
@@ -0,0 +1,161 @@
+/*
+ * GridTools
+ *
+ * Copyright (c) 2014-2021, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/**
+ * @file
+ *
+ * Implements operations on `int_vector`s, which are hymaps of integers (integral types or integral_constants)
+ *
+ * A type `T` is `int_vector` if
+ *  - `is_hymap<T>`
+ *  - all value types `V` are std::is_integral<V> or is_gr_integral_constant<V>
+ *  - `std::is_trivially_copy_constructible<T>` // TODO isn't that trivially satisfied by the previous conditions?
+ */
+
+#pragma once
+
+#include <type_traits>
+
+#include "../meta.hpp"
+#include "defs.hpp"
```

Comment:
not sure it is used

Reply:
GT_FUNCTION is coming from there.

Reply:
Ah no, but GT_CONSTEXPR. Which reminds me of the question that I had yesterday: is GT_CONSTEXPR still a thing? I.e. does nvcc still not understand `constexpr` for the version we are supporting?

Reply:
`nvcc` did `constexpr` wrong in several ways in several versions. 
There were two aspects that I remember:
  - sometime run time execution of `constexpr` in device degrade performance.
  - sometime `nvcc` has compilation errors. mostly when the `constexpr` functions are part of expressions in non evaluation  context (`decltype`).

Hence to check this again with CUDA 11 you need full test suite (unit tests + perfs)